src/client/mod.rs

```diff
@@ -79,37 +85,142 @@ impl<'a> FobnailClient<'a> {
+                }
+            }
+            State::RequestAik {
+                ref mut request_pending,
+                ..
+            } => {
+                if !*request_pending {
+                    *request_pending = true;
+                    let mut request = coap_lite::CoapRequest::new();
+                    request.set_path("/aik");
+                    request.set_method(RequestType::Fetch);
+                    let state = Rc::clone(&self.state);
+                    self.coap_client
+                        .queue_request(request, move |result| Self::handle_response(result, state));
+                }
+            }
+            State::VerifyAikStage1 { data, ek_cert } => {
+                let mut trussed = self.trussed_platform.borrow_mut();
+
+                match Self::decode_aik(&mut *trussed, &data).and_then(
+                    |(_tpm_public, loaded_key_name)| {
+                        Self::prepare_aik_challenge(
+                            &mut *trussed,
+                            tpm::mu::LoadedKeyName::decode(&loaded_key_name).unwrap(),
+                            ek_cert,
+                        )
+                    },
+                ) {
+                    Ok((secret, id_object, encrypted_secret)) => {
+                        // Trick to move data without copying.
+                        // Rust won't allow us just move out data because doing
+                        // so would make state invalid.
+                        let mut aik = vec![];
+                        core::mem::swap(data, &mut aik);
+
+                        *state = State::VerifyAikStage2 {
+                            request_pending: false,
+                            secret,
+                            id_object,
+                            encrypted_secret,
+                            aik,
+                        }
+                    }
+                    Err(()) => {
+                        error!("AIK verification failed");
+                        *state = State::Idle {
+                            timeout: Some(get_time_ms() as u64 + 5000),
+                        };
+                    }
+                }
             }
+            State::VerifyAikStage2 {
+                request_pending,
+                id_object,
+                encrypted_secret,
+                ..
+            } => {
+                if !*request_pending {
+                    let encoded = trussed::cbor_serialize_bytes::<_, 512>(&proto::Challenge {
+                        id_object: &id_object,
+                        encrypted_secret: &encrypted_secret,
+                    })
+                    .unwrap();
+
+                    let mut request = coap_lite::CoapRequest::new();
+                    request.set_path("/challenge");
+                    request.set_method(RequestType::Post);
+                    request.message.payload = encoded.to_vec();
+                    // TODO: should set Content-Type header to CBOR
```

Comment:
```suggestion
                    request.message.set_content_format(coap_lite::ContentFormat::ApplicationCBOR);
```

This isn't checked on Attester side right now, but at some point we may depend on it.

Reply:
Fixed